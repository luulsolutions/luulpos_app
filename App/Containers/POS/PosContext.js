import React from 'react'

const PosContext = React.createContext({
	inCategory: true,
	isModalVisible: false,
	searchTerm: '',
	categoryId: null,
	setCategoryId: (id) => {},
	setInCategory: (value) => {},
	peformSearch: (searchTerm) => {},
	setModalVisible: (value) => {},
	cancelSearch: () => {}
})

export default PosContext
