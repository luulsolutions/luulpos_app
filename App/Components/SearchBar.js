import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, TextInput, TouchableOpacity } from 'react-native';
import styles from './Styles/SearchBarStyles';
import { Colors, Metrics } from '../Themes/';
import Icon from 'react-native-vector-icons/FontAwesome';
import PosContext from '../Containers/POS/PosContext';
import Icons from 'react-native-vector-icons/MaterialIcons';
import colors from '../Themes/Colors';

export default class SearchBar extends React.Component {
  render() {
   // const { onSearch, onCancel, searchTerm } = this.props;
    return (
      <PosContext.Consumer>
        {value => (
          <View style={styles.container}>
            <Icon name="search" size={Metrics.icons.tiny} style={styles.searchIcon} />
            <TextInput
              placeholder="Search"
              placeholderTextColor={Colors.snow}
              underlineColorAndroid="transparent"
              style={styles.searchInput}
              value={value.searchTerm}
              onChangeText={value.peformSearch}
              autoCapitalize="none"
              onSubmitEditing={value.peformSearch.bind(this, value.searchTerm)}
              returnKeyType={'search'}
              autoCorrect={false}
              selectionColor={Colors.snow}
            />
            <TouchableOpacity onPress={value.cancelSearch} style={styles.cancelButton}>
              <Text style={styles.buttonLabel}><Icons name="clear"  color={Colors.primary} size={Metrics.icons.tiny}/></Text>
            </TouchableOpacity>
          </View>
        )}
      </PosContext.Consumer>
    );
  }
}
