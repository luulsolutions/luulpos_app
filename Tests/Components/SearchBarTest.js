import SearchBar from '../../App/Components/SearchBar'
import PosContext from '../../App/Containers/POS/PosContext'
import React from 'react'

jest.mock(SearchBar, () => {
  const ThemeProvider = require.requireActual('../../App/Containers/POS/PosContext')
  const theme = require.requireActual('../../App/Components/SearchBar')

  return {
    ThemeProvider,
    ThemeConsumer: (props) => props.children({ theme })
  }
})

test('Should find the Text component', () => {
  const wrapper = shallow(<SearchBar />)

  expect(wrapper.find('ThemeConsumer').dive().find('[searchTerm]').length).toEqual(1)
})
