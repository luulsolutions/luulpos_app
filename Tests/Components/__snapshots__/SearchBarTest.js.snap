// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0)",
      "flexDirection": "row",
      "marginTop": 5,
      "width": 375,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 15,
        },
        Object {
          "alignSelf": "center",
          "backgroundColor": "rgba(0,0,0,0)",
          "color": "white",
          "left": 20,
        },
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï€‚
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Search"
    placeholderTextColor="white"
    returnKeyType="search"
    selectionColor="white"
    style={
      Object {
        "alignSelf": "center",
        "borderColor": "white",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "white",
        "flex": 1,
        "flexDirection": "row",
        "fontFamily": "Avenir-Book",
        "fontSize": undefined,
        "height": 30,
        "marginLeft": 1,
        "marginRight": 10,
        "padding": 5,
        "paddingLeft": 30,
        "textAlign": "left",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginHorizontal": 10,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Avenir-Book",
          "fontSize": 17,
        }
      }
    >
      Cancel
    </Text>
  </View>
</View>
`;
